aws_lambda_builders/workflow.py

```diff
@@ -4,28 +4,78 @@
 
 import os
 import logging
+
+from collections import namedtuple
 import six
 
-from aws_lambda_builders.registry import WorkflowMetaClass
+from aws_lambda_builders.registry import DEFAULT_REGISTRY
+from aws_lambda_builders.exceptions import WorkflowFailedError, WorkflowUnknownError
+from aws_lambda_builders.actions import ActionFailedError
 
 LOG = logging.getLogger(__name__)
 
 
-class BaseWorkflow(six.with_metaclass(WorkflowMetaClass, object)):
+# Named tuple to express the capabilities supported by the builder.
+# ``Language`` is the programming language. Ex: Python
+# ``LangageFramework`` is the framework of particular language. Ex: PIP
```

Comment:
Would PackageFramework be better?  LanguageFramework confused me initially into thinking it meant ApplicationFramework.  Maybe just me.

Reply:
Or maybe just Packager?

Reply:
Good point. I'll call it `DependencyFramework` then..

Reply:
I went thru a websites for whole bunch of dependency frameworks like PIP, Yarn, NPM, Bundle etc. `DependencyManager` is the closest consistent definition